servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -567,12 +570,12 @@ void RendererSceneRenderRD::_render_buffers_post_process_and_tonemap(const Rende
 			dest_fb = texture_storage->render_target_get_rd_framebuffer(render_target);
 		}
 
-		tone_mapper->tonemapper(internal_texture, dest_fb, tonemap);
+		tone_mapper->tonemapper(color_texture, dest_fb, tonemap);
 
 		RD::get_singleton()->draw_command_end_label();
 	}
 
-	if (fsr && can_use_effects && rb->get_scaling_3d_mode() == RS::VIEWPORT_SCALING_3D_MODE_FSR) {
+	if (use_fsr) {
```

Comment:
btw, on a foot note, we should do this in more places in the code, instead of copying a complex check in multiple points, just set a variable as early on and use that variable. Makes code a lot easier to read :)

Reply:
Yup, I'm a big fan of declaring boolean variables where the name is descriptive of what the condition means instead of throwing it all in the if clause. It tends to save on needing to have a comment describing what the condition is as well, and it should get optimized by the compiler fairly easily.